src/opensynth/models/faraday/model.py

```diff
@@ -499,33 +511,47 @@ def train_gmm(self, dm: LCLDataModule):
         Args:
             dm (LCLDataModule): Training data
         """
+        logger.info("🚀 Training GMM")
+
         dl = dm.train_dataloader()
         next_batch = next(iter(dl))
 
         # Explicit check to make sure that data module used
         # to train GMM has features ordered the same way as
         # when training the VAE
-        obtained_feature_list = list(next_batch["features"].keys())
+        features = next_batch["features"]
+        obtained_feature_list = list(features.keys())
         expected_feature_list = self.vae_module.feature_list
-        assert obtained_feature_list == expected_feature_list
-
-        for batch_num, batch_data in tqdm(enumerate(dl)):
-            kwh = batch_data["kwh"]
-            features = batch_data["features"]
-
-            vae_input = self.vae_module.reshape_data(kwh, features)
-            vae_output = self.vae_module.encode(vae_input)
+        if obtained_feature_list != expected_feature_list:
+            logger.error(
+                """Feature list required by `vae_module` and does not match
+                features specified by the data module.
+                """
+            )
 
-            gmm_input = self.vae_module.reshape_data(vae_output, features)
-            self.gmm.fit(gmm_input.detach().numpy())
-            logger.info(f"⏳ Batch {batch_num} completed")
+        gmm_module, _, _ = fit_gmm(
```

Comment:
we only use the `gmm_module` returned from `fit_gmm`, so don't need to assign the other returned objects